src/predictor/gpu_predictor.cu

```diff
@@ -246,11 +269,13 @@ class GPUPredictor : public xgboost::Predictor {
     void PredictInternal
     (const SparsePage& batch, const MetaInfo& info,
      HostDeviceVector<bst_float>* predictions,
-     const size_t batch_offset,
      const gbm::GBTreeModel& model,
      const thrust::host_vector<size_t>& h_tree_segments,
      const thrust::host_vector<DevicePredictionNode>& h_nodes,
      size_t tree_begin, size_t tree_end) {
+      if (predictions->DeviceSize(device_) == 0) {
```

Comment:
the couple of lines from line# 296 can be hoisted out and moved here before the trees are copied to the gpu.

Reply:
Hmm, the offsets are sharded using the standard `Overlap` distribution. Looks like that doesn't handle the last batch correctly. This check should be equivalent.

Reply:
oh - what i meant was to add that check also here and bail out early, as opposed to building the tree on gpu, check and bailing out. sorry for the super unhelpful comment earlier.

Reply:
The second check is actually no longer needed. Removed.